lib/fastifySession.js

```diff
@@ -218,7 +220,7 @@ function shouldSaveSession (request, cookieOpts, saveUninitialized) {
 }
 
 function isSessionModified (session) {
-  return (Object.keys(session).length !== 4)
+  return (Object.keys(session).length !== 3)
```

Comment:
Session is not a plain JavaScript object. It is an instance of https://github.com/fastify/session/blob/73a4675b795e8ad3d020d5cd43de78374c09f8aa/lib/session.js#L12

That object should provide a `.keys()` method to get this information correctly.